src/utils/disable-ios-zoom.js

```diff
@@ -0,0 +1,24 @@
+import MobileDetect from "mobile-detect";
+const mobiledetect = new MobileDetect(navigator.userAgent);
+
+export function disableiOSZoom() {
+  if (!mobiledetect.is("iPhone") && !mobiledetect.is("iPad")) return;
+
+  let lastTouchAtMs = 0;
+
+  document.addEventListener("touchmove", ev => {
+    if (ev.scale === 1) return;
+
+    ev.preventDefault();
+  });
+
+  document.addEventListener("touchend", ev => {
+    const now = new Date().getTime();
+    const isDoubleTouch = now - lastTouchAtMs <= 300;
```

Comment:
I'm pretty sure, but not certain, you could use iOS' [`gesturestart`/`gesturechange`/`gestureend` events](https://developer.apple.com/library/content/documentation/AppleApplications/Reference/SafariWebContent/HandlingEvents/HandlingEvents.html).

from the [Safari's _Handling Events_ https://developer.apple.com/library/content/documentation/AppleApplications/Reference/SafariWebContent/HandlingEvents/HandlingEvents.html
> To disable pinch open and pinch close gestures in iOS 2.0, implement the `gesturestart` and `gesturechange` event handlers as follows:

```
function gestureChange(event) {
  // Disable browser zoom
  event.preventDefault();
}
```

(there's also [`dblclick`](https://developer.mozilla.org/en-US/docs/Web/Events/dblclick) but that may fire too late.)

Reply:
yeah the double tapping is the main issue for the link page, and the zooming is the main issue for the hub page. this implementation "cleanly" fixes both (I don't like it either) :/